kdTree.cpp

```diff
@@ -2,45 +2,46 @@
 #include "kdTree.h"
 
 namespace Tmpl8 {
-	kdTree::kdTree()
+	KdTree::KdTree()
 	{
 
 	}
 
+    struct node
+    {
+        Tank* tank;
+        struct node* left;
+        struct node* right;
+
+    };
+
     /// <summary>
     /// creates a new node
     /// </summary>
     /// <param name="tank"> a pointer to a tank </param>
     /// <returns>returns a pointer to the just made node</returns>
-    kdTree::node* kdTree::newnode(Tank* tank)
+    KdTree::node* KdTree::newnode(Tank* tank)
 	{
-
-        struct node* node
-            = (struct node*)malloc(sizeof(struct node));
-
-        node->tank = tank;
-        node->left = NULL;
-        node->right = NULL;
-
-        return node;
+        struct node* temp = new node;
```

Comment:
```new``` keyword allocates the memory on the heap which is slower in theory. For the size of our data it would be fine to allocate on the stack instead by using the constructor instead of the new keyword. This isn't really a big deal though because I doubt the performance changes are that significant. Just something useful to know about c++.

Reply:
After discussion we decided that using new is fine because the performance benefits are marginal and we want the nodes to persist. Technically this leaks memory and isn't good practice but in our specific use case we don't care.